llm/payload.go

```diff
@@ -0,0 +1,223 @@
+package llm
+
+import (
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/jmorganca/ollama/gpu"
+)
+
+var errPayloadMissing = fmt.Errorf("expected payloads not included in this build of ollama")
+
+func Init() error {
+	payloadsDir, err := gpu.PayloadsDir()
+	if err != nil {
+		return err
+	}
+
+	slog.Info("extracting embedded files", "dir", payloadsDir)
+
+	// also extract the .metal shader on darwin/arm64
+	if runtime.GOOS == "darwin" && runtime.GOARCH == "arm64" {
```

Comment:
Do we still need this? We can compile the .metal file in with `LLAMA_METAL_EMBED_LIBRARY`